code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -884,6 +887,86 @@
 		new /obj/item/stack/cable_coil(loc, 3)
 	qdel(src)
 
+/obj/item/circuit_component/air_alarm
+	display_name = "Air Alarm"
+	display_desc = "Controls levels of gases and their temperature as well as all vents and scrubbers in the room."
+
+	var/datum/port/input/min_2
+	var/datum/port/input/min_1
+	var/datum/port/input/max_1
+	var/datum/port/input/max_2
+
+	var/datum/port/input/request_data
+
+	var/datum/port/output/pressure
+	var/datum/port/output/temperature
+	var/datum/port/output/gas_amount
+
+	var/obj/machinery/airalarm/connected_alarm
+	var/list/options_map
+
+/obj/item/circuit_component/air_alarm/Initialize()
+	. = ..()
+	min_2 = add_input_port("Min 2", PORT_TYPE_NUMBER)
+	min_1 = add_input_port("Min 1", PORT_TYPE_NUMBER)
+	max_1 = add_input_port("Max 1", PORT_TYPE_NUMBER)
+	max_2 = add_input_port("Max 2", PORT_TYPE_NUMBER)
+	request_data = add_input_port("Request Atmosphere Data", PORT_TYPE_SIGNAL)
+
+	pressure = add_output_port("Pressure", PORT_TYPE_NUMBER)
+	temperature = add_output_port("Temperature", PORT_TYPE_NUMBER)
+	gas_amount = add_output_port("Chosen Gas Amount", PORT_TYPE_NUMBER)
+
+/obj/item/circuit_component/air_alarm/populate_options()
+	var/static/list/component_options = list(
+		"Pressure",
+		"Temperature"
+	)
+	options = component_options
+
+	var/static/list/options_to_key = list(
+		"Pressure" = "pressure",
+		"Temperature" = "temperature"
+	)
+
+	for(var/gas_type in subtypesof(/datum/gas))
+		var/datum/gas/new_gas = new gas_type()
+		component_options.Add(new_gas.name)
+		options_to_key[new_gas.name] = gas_type
+		qdel(new_gas)
+
+	options_map = options_to_key
+
+/obj/item/circuit_component/air_alarm/register_usb_parent(atom/movable/parent)
+	. = ..()
+	if(istype(parent, /obj/machinery/airalarm))
+		connected_alarm = parent
+
+/obj/item/circuit_component/air_alarm/unregister_usb_parent(atom/movable/parent)
+	connected_alarm = null
+	return ..()
+
+/obj/item/circuit_component/air_alarm/input_received(datum/port/input/port)
+	. = ..()
+
+	if(. || !connected_alarm || connected_alarm.locked)
+		return
+
+	if(COMPONENT_TRIGGERED_BY(request_data, port))
+		var/turf/alarm_turf = get_turf(connected_alarm)
+		var/datum/gas_mixture/environment = alarm_turf.return_air()
+		pressure.set_output(round(environment.return_pressure()))
+		temperature.set_output(round(environment.temperature))
+		if(ispath(options_map[current_option]))
+			gas_amount.set_output(round(environment.gases[options_map[current_option]][MOLES]))
+		return
+
+	var/datum/tlv/tlv = connected_alarm.TLV[options_map[current_option]]
```

Comment:
I don't know this acronym, I know it's used by other stuff but can you expand it?

Reply:
Finding the source of this is suffering, paging @Kylerace

Reply:
threshold limit values, the comment explaining this was removed somewhere along the way but it used to be https://github.com/Kylerace/tgstation/blob/c2c7a3bcda38011e1c9a345ebd7c2a9225c557a8/code/game/machinery/alarm.dm

Reply:
lmao